src/app/ConcreteAttributePath.h

```diff
@@ -113,7 +113,7 @@ struct ConcreteDataAttributePath : public ConcreteAttributePath
 
     ConcreteDataAttributePath() {}
 
-    ConcreteDataAttributePath(const ConcreteAttributePath & path) : ConcreteAttributePath(path) {}
+    explicit ConcreteDataAttributePath(const ConcreteAttributePath & path) : ConcreteAttributePath(path) {}
```

Comment:
This seem to cause a few `ConcreteDataAttributePath(` additions which feels noisy. Could we fix the operator== instead?

Reply:
I've tried. It just adds more ambiguity, not just for compiler but for those who will read the code. And more places for potential bugs.
I've eventually fixed concrete `operator==` calls. There are few of them

Reply:
We could add  `operator==(const ConcreteAttributePath &)` maybe? 

Tricky as that may differ from today: do we want to ignore the data path element (what we have today, but it is unintuitive) or assert those are defaults/not-a-list-element (does that break backwards compatibility?)

Reply:
The problem is that for now the result differs depending on order. So:
```cpp
ConcreteAttributePath aPath;
ConcreteDataAttributePath bPath;

(aPath == bPath) != (bPath == aPath)
```

Because:
- `aPath == bPath` will use `ConcreteAttributePath::operator==` with implicit slicing of `bPath` to `ConcreteAttributePath`. And will skip the list and other stuff comparison
- `bPath == aPath` will use `ConcreteDataAttributePath::operator==` with implicit conversion of `aPath` to `ConcreteDataAttributePath`. And will probably will almost always be `false` as `bPath` will have default values in `mListIndex` and other fields

Explicit `operator==(const ConcreteAttributePath &)` will definitely break this behavior. But it seems like it is not expected behavior.